__tests__/src/integration.spec.js

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2017 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+const fs = require('fs');
+const path = require('path');
+const toMatchImageSnapshot = require('../../src').toMatchImageSnapshot;
+
+expect.extend({ toMatchImageSnapshot });
+
+describe('integration tests', () => {
+  const imagePath = path.resolve(__dirname, 'TestImage.png');
+  const data = fs.readFileSync(imagePath);
+
+  it('matches an identical (byte-for-byte) snapshot.', () => {
```

Comment:
Not sure I understand the difference between these 2 tests. Can you elaborate?

Reply:
Byte-for-byte compares two .PNG files that are exactly the same, while pixel-for-pixel compares two .PNG files that resolve to the same image.  For this specific test, it's comparing two .PNG files of the same image, but different compression ratios.  Since `jest-image-snapshot` works on .PNG files instead of bitmaps, it's a pretty brittle test, but I figured it was worth including.

Reply:
But the test image is the same though? Or did you change the compression ratio on the snapshot?

Reply:
The snapshot.  This is why I say this is a brittle test: the snapshot needs to be manually created, since `jest-image-snapshot` saves the .PNG file.
This and the lack of `not` support has been bothering me in regards to this PR.  I believe that integration tests are vital to things like this, but I am dissatisfied with the solution that I have proposed.

Reply:
I agree an integration test is necessary so maybe what we do is just have the one integration test for now (`it('matches an identical', () =>...`) just to validate the "happy path."

For a separate PR (or if you want to add it here) what could be useful as well are tests that validate that snapshot is generated at correct path by `fs` and whatnot to improve integration test coverage.

Reply:
Sorry for the delay, it was a long weekend.
I actually had a question about that.  Why do image snapshots go to their own folder instead of the `__snapshots__` folder?

Reply:
Mostly because I didn’t want to interfere with or be interfered with by anything ‘jest’ may be doing or may start doing in the future to clean up snapshots in that directory.

Reply:
That makes sense.
I'd like to split up `diff-snapshot.js` before adding that test.

Reply:
alright so then were good to merge this?

Reply:
Yeah.  If you'll have it, I'll see if I can make a PR to split some stuff soon.